linkerd/app/outbound/src/tcp/opaque_transport.rs

```diff
@@ -80,6 +83,8 @@ where
             }
         }
 
+        let protocol: Option<SessionProtocol> = ep.param();
```

Comment:
this makes me wonder how possible it would be to add a `Parse`-style handshake trait for `Param`, so you can write
```rust
let protocol = ep.get_param::<Option<SessionProtocol>>();
```
instead. not a blocker, but it might make some code easier to write such as when a param is part of a method chain

Reply:
Seems like basically the same amount of typing, so i'm -0 on it for now.